dev/system-test/firestore.ts

```diff
@@ -992,6 +992,19 @@ describe('DocumentReference class', () => {
       unsubscribeCallbacks.forEach(c => c());
     });
 
+    it('waits for existing operations to complete before releasing clients', async () => {
+      const ref = randomCol.doc('doc-1');
+
+      const unsubscribe = ref.onSnapshot(() => {
+        // No-op
+      });
+
+      await ref.set({});
+      unsubscribe();
+      // Terminate should allow unsubscribe to complete.
+      await firestore.terminate();
```

Comment:
I think this test won't actually fail in the current client, it will just log a rejected Promise warning. We might be better off adding this as a unit test to the ClientPool unit tests.